File: donghwa/0x06/10845.py
import sys
N = int(sys.stdin.readline().strip())
cmds = [sys.stdin.readline().strip() for _ in range(N)]
queue = []
for cmd in cmds :
    if cmd[:4] == "push":
        queue.append(cmd.split()[-1])
    elif cmd == "pop":
        if len(queue) > 0 :
            print(queue.pop(0))
        else :
            print(-1)
    elif cmd == "size":
        print(len(queue))
    elif cmd == "empty":
        if len(queue) == 0:
            print(1)
        else :
            print(0)
    elif cmd == "front":
        if len(queue) > 0 :
            print(queue[0])
        else :
            print(-1)
    elif cmd == "back":
        if len(queue) > 0 :
            print(queue[-1])
        else :
            print(-1)
    N -= 1

File: donghwa/0x06/10845.cpp
#include <bits/stdc++.h>
using namespace std;

int main(){
    ios::sync_with_stdio(0);
    cin.tie(0);

    queue<int> Q;
    int N;
    cin >> N;
    while (N--){
        string cmd;
        cin >> cmd;
        if(cmd=="push"){
            int x;
            cin >> x;
            Q.push(x);
        } else if(cmd=="pop"){
            if (Q.empty()){
                cout << "-1\n";
            } else {
                cout << Q.front() << '\n';
                Q.pop();
            }
        } else if(cmd=="size"){
            cout << Q.size() << '\n';
        } else if(cmd=="empty"){
            cout << Q.empty() << '\n';
        } else if(cmd=="front"){
            if (Q.empty()) cout << "-1\n";
            else cout << Q.front() << '\n';
        } else if(cmd=="back"){
            if (Q.empty()) cout << "-1\n";
            else cout << Q.back() << '\n';
        }
    }
}